module/x/gravity/types/batch.go

```diff
@@ -1,21 +1,21 @@
 package types
 
 import (
+	fmt "fmt"
 	"math/big"
 	"strings"
 
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/ethereum/go-ethereum/accounts/abi"
 	gethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
 // GetCheckpoint gets the checkpoint signature from the given outgoing tx batch
-func (b OutgoingTxBatch) GetCheckpoint(gravityIDstring string) ([]byte, error) {
+func (b OutgoingTxBatch) GetCheckpoint(gravityIDstring string) []byte {
```

Comment:
are we expecting these checkpoints to panic now?

Reply:
The only way they will panic is through programmer error, which means that it is unnecessary to handle it at runtime. One of the GetCheckpoint functions was already panicking, while the other two were erroring as you see here. When unifying the functions into a single interface, I looked at the code and made a judgement call to use panics. If I am wrong, and there is some way that the errors could happen depending on different conditions at runtime, then we should switch them all to errors.